library/Freckle/App.hs

```diff
@@ -59,6 +59,13 @@
 -- >     myAppAction
 -- >     myOtherAppAction
 --
+-- == 'AppT'
```

Comment:
Is it possible to avoid exposing the stack so that clients are forced to use the constraints?

Reply:
Anything is possible!

I can not export `AppT(..)` or `runAppT`. Then users would be unable to do the things I mentioned. They'd be forced to use constraints and only ever call `runApp`.

I dislike it when a function (`runApp`) mentions a type (`AppT`) that isn't exported though because:

- You can't get that type's Haddocks (minor)
- You can't extract/use the function in a way that requires a type signature (medium)

I could solve these by exporting `AppT`, but not `(..)` or `runAppT`. Then it's just about whether the limitations are a feature or a bug. What if they needed to use this `AppT` with a different `m` than `IO`? We'd be disallowing that valid use in an attempt to forbid invalid use.

But then, what use are we forbidding that's invalid? There can be no _current_ uses, since `AppT` doesn't exist. And if there are uses that are explicitly `ReaderT`, they'll be broken if we export `AppT` or not. So we'd be guarding against future explicit uses of `AppT`, in case we want to change it -- which is what I mentioned in the comment.

But, wait, is that event true? I think we _can_ evolve the stack under `AppT` while also exporting it: `AppT(..)` + `runAppT` won't change if we do such things. And, even if it did, we play really loose with compatibility in freckle-app since it's not a core library and is more of a "leaf" package anyway.

So I think I just want to ditch the comment hinting at this being any kind of "internal" thing.